### PR TITLE
Renew After Expiry

### PR DESCRIPTION
**Pull request by maraino**
### Description
This PR adds support for certificate renewal after expiry. Provisioners with the claim `"allowRenewAfterExpiry": true` can send an Authorization header with an x5c-like token. After validating the token the certificate can be renewed.

### CODE REVIEW

File: api/api.go [hunk 1]
```diff
@@ -33,6 +33,7 @@ type Authority interface {
 	// context specifies the Authorize[Sign|Revoke|etc.] method.
 	Authorize(ctx context.Context, ott string) ([]provisioner.SignOption, error)
 	AuthorizeSign(ott string) ([]provisioner.SignOption, error)
+	AuthorizeRenewToken(ctx context.Context, token string) (*x509.Certificate, error)
```
[the remark this answers is not in the document]
Comment (maraino) [reply api/api.go]: Fixed with f8df6a1

File: authority/provisioner/extension.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+	"encoding/asn1"
+)
+
+var (
+	// StepOIDRoot is the root OID for smallstep.
+	StepOIDRoot = asn1.ObjectIdentifier{1, 3, 6, 1, 4, 1, 37476, 9000, 64}
+
+	// StepOIDProvisioner is the OID for the provisioner extension.
+	StepOIDProvisioner = append(asn1.ObjectIdentifier(nil), append(StepOIDRoot, 1)...)
+)
```
Comment (hslatman) [inline authority/provisioner/extension.go:15]: I like it.


Other projects may be able to use this. Not sure if needs to be in a separate package (yet). `certinfo` can probably use this.

File: ca/testdata/ca.json [hunk 3]
```diff
@@ -6,7 +6,7 @@
     "password": "password",
     "address": "127.0.0.1:0",
     "dnsNames": ["127.0.0.1"],
-    "logger": {"format": "text"},
+    "_logger": {"format": "text"},
```
Comment (hslatman) [inline ca/testdata/ca.json:9]: I guess these disable logging for tests? If that's indeed the goal, then it's OK 🙂

**Review by hslatman — APPROVED** [review]

File: ca/testdata/ca.json (hunk 3 above)
Comment (maraino) [reply ca/testdata/ca.json:9]: Yes, it's hard to see what's going on if something fails. You can easily enable it for debugging purposes locally.

**Pull request merged**